File: PlacementCellBackend/Controllers/CompanyController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.Models;
using PlacementCellBackend.Services.CRUD.Interfaces;

namespace PlacementCellBackend.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class CompanyController : ControllerBase
    {
        private readonly ICompanyService _companyService;

        public CompanyController(ICompanyService companyService)
        {
            _companyService = companyService;
        }

        [HttpGet]
        public async Task<ActionResult<IEnumerable<Company>>> GetCompany()
        {
            var companies = await _companyService.GetAllCompaniesAsync();
            return Ok(companies);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<Company>> GetCompany(string id)
        {
            var company = await _companyService.GetCompanyByIdAsync(id);
            if (company == null)
                return NotFound();
            return Ok(company);
        }

        [HttpPost]
        public async Task<ActionResult<Company>> PostCompany(Company company)
        {
            var created = await _companyService.CreateCompanyAsync(company);
            return CreatedAtAction(nameof(GetCompany), new { id = created.company_id }, created);
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> PutCompany(string id, Company updatedCompany)
        {
            if (id != updatedCompany.company_id)
                return BadRequest();

            var success = await _companyService.UpdateCompanyAsync(id, updatedCompany);
            if (!success)
                return NotFound();

            return NoContent();
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteCompany(string id)
        {
            var success = await _companyService.DeleteCompanyAsync(id);
            if (!success)
                return NotFound();

            return NoContent();
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/CRUD/CompanyController.java
package com.example.placementicsbackend.controllers.CRUD;

import com.example.placementicsbackend.models.Company;
import com.example.placementicsbackend.services.CRUD.Interfaces.ICompanyService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/company")
@RequiredArgsConstructor
public class CompanyController {

    private final ICompanyService companyService;

    @GetMapping
    public ResponseEntity<List<Company>> getCompany() {

        List<Company> companies = companyService.getAllCompanies();

        return ResponseEntity.ok(companies);
    }

    @GetMapping("/{id}")
    public ResponseEntity<Company> getCompany(
            @PathVariable String id) {

        Company company = companyService.getCompanyById(id);

        if (company == null) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.ok(company);
    }

    @PostMapping
    public ResponseEntity<Company> postCompany(
            @RequestBody Company company) {

        Company created = companyService.createCompany(company);

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(created);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Void> putCompany(
            @PathVariable String id,
            @RequestBody Company updatedCompany) {

        if (!id.equals(updatedCompany.getCompanyId())) {
            return ResponseEntity.badRequest().build();
        }

        boolean success =
                companyService.updateCompany(id, updatedCompany);

        if (!success) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.noContent().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteCompany(
            @PathVariable String id) {

        boolean success = companyService.deleteCompany(id);

        if (!success) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.noContent().build();
    }
}